packages/ketcher-polymer-editor-react/src/state/rna-builder/rnaBuilderSlice.ts

```diff
@@ -110,11 +110,13 @@ export const rnaBuilderSlice = createSlice({
     deletePreset: (state, action: PayloadAction<IRnaPreset>) => {
       const preset = action.payload;
 
+      const presetIndexInList = state.presets.findIndex(
+        (presetInList) => presetInList.name === preset.name,
+      );
+      state.presets.splice(presetIndexInList, 1);
+
       if (preset.presetInList) {
-        const presetIndexInList = state.presets.findIndex(
-          (presetInList) => presetInList.name === preset.presetInList?.name,
-        );
```

Comment:
Do we need to change the logic in savePreset too? because a new created preset will be saved in `presetInList` of activePreset and it seems could not be deleted if we dont use `presetInList?.name` to match here